### PR TITLE
Optimize logic of retrieving user agent appID from env config

### PR DESCRIPTION
**Pull request by wty-Bryant**
In a previous [PR](https://github.com/aws/aws-sdk-go-v2/pull/2154),  a new `appID` field is introduced into request's user agent header. In that old PR, `appID` set via env var is retrieved separately after iterating  resolver chain's all configs, this pr now optimize this step by setting `appID` as a field of env config so that it be retrieved directly from env config struct

### CODE REVIEW

File: config/env_config.go
Comment (isaiahvita) [inline config/env_config.go:339]: while this works, i dont think this is the typical setup. see the other fields in `env_config.go`

Comment (wty-Bryant) [reply config/env_config.go:339]: Just want to lazy set this field in case it is modified, but you are right, I will change that

**Comment by isaiahvita** [comment]
@wty-Bryant also, while i already know the context of this PR. it will be good to clearly explain the motivation for it in the description

File: config/env_config.go
Comment (isaiahvita) [reply config/env_config.go:339]: i dont think there is actually any reasonable time interval between when `NewEnvConfig` is called and when `getAppId` is called. so i dont think lazily evaluating it here makes a difference

Comment (lucix-aws) [reply config/env_config.go:339]: Yeah - `loadEnvConfig` first reads everything out of the environment into the env config struct, and then the "resolvers" e.g. `getAppID` just read out of the struct. `getAppID` shouldn't be querying the environment directly. See other fields in `env_config` as @isaiahvita said.

Comment (lucix-aws) [reply config/env_config.go:339]: > Just want to lazy set this field in case it is modified,

> i dont think there is actually any reasonable time interval

There's no change of changing or interval to speak of - a process' environment is static.

**Review by isaiahvita — APPROVED** [review]
LGTM, wait until the relevant CI passes though before merging

**Review by lucix-aws — APPROVED** [review]

**Pull request merged**
